docs/csharp/language-reference/operators/switch.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "switch expressions - C# reference"
+description: Learn how to use the C# switch expression for pattern matching and other data introspection
+ms.date: 03/19/2020
+---
+# switch expressions (C# reference)
+
+This article covers the `switch` expression, introduced in C# 8.0. For information on the `switch` statement, see the article on the [`switch` statement](../statements/switch.md) in the [statements](../statements/index.md) section.
+
+## Basic example
+
+The `switch` expression provides for `switch`-like semantics in an expression context. It provides a concise syntax when the switch arms produce a value. The following example shows the structure of a switch expression. It translates values from an `enum` representing visual directions in an online map to the corresponding cardinal direction:
+
+:::code language="csharp" source="snippets/SwitchExpressions.cs" id="SnippetBasicStructure" interactive="csharp-interactive":::
```

Comment:
Is it the new interactive system in action? The snippet doesn't include `enum` definitions, but will be runnable anyway?

Reply:
The new system is deployed. I'm making all the snippets in this article non-interactive because the new system uses scriptcs, which doesn't have full fidelity with C# 8.0 features.